### PR TITLE
refactor: restructure rss feed table

### LINKED ISSUE #722: Restructure homepage release notes into one table
All release notes on the homepage for all microsites should be listed in one table.
Each entry should also contain the microsite where it's coming from.

Suggestion:

| Release Notes  |   |   |
|---|---|---|
| 26 October  | Merchant Center  | "Release note text"  |
| 20 October  | HTTP API  | "Release note text"  |
| 15 October  | HTTP API  | "Release note text"  |
| 12 October  | Import API | "Release note text"  |
etc...

### PR DESCRIPTION
**Pull request by TimonRey**
resolves #722 

**Hints for review**
- After thinking it over again, I decided to pass the feed url over props to the component. I think it makes more sense than to wrap the feeds with a parent component.
- I seperated the logic into two components called `RssFeed` which handles the requests and refactores the data, and `RssFeedTable` which gets the data from `RssFeed` and simply puts them into the table

### CODE REVIEW

File: packages/ui-kit/src/components/rss-feeds.js
Comment (TimonRey) [inline packages/ui-kit/src/components/rss-feeds.js:51]: If you're wondering why we need this logic, take a look at the ticket to see the requirements
https://github.com/commercetools/commercetools-docs-kit/issues/722#issuecomment-718179010

**Comment by davifantasia** [comment]
This table looks a bit confusing from the description of the issue. So just I can be sure, is this the UI Adina/Sven expected? One of the problems is that there is just one release note from each product.

Screenshot 2020-11-03 at 08 55 11

File: packages/ui-kit/src/components/rss-feeds.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.js:59]: Can't we combine the 3 reduce functions into one?

File: packages/ui-kit/src/components/rss-feeds.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.js:20]:
```suggestion
  return Promise.all(promises);
```

File: packages/ui-kit/src/components/rss-feeds.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.js:69]: I would prefer to do the "transformation/aggregation" of the data into a separate function, to not bloat too much the component.

File: packages/ui-kit/src/components/rss-feeds.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.js:75]: Also here, specify the shape of the objects.

**Review by emmenko — CHANGES_REQUESTED** [review]
Thanks, looks good in general! We can improve a bunch of things though. 🤗

File: packages/gatsby-theme-docs/gatsby-config.js
Comment (TimonRey) [reply packages/gatsby-theme-docs/gatsby-config.js:265]: Actually, the idea was to show more release notes in the table, but i'm not sure if we really need more release notes. I think the list is long enough...at the moment there are 9 entries and in "worst case", there would be only 5 release notes shown in the table, so I think we just have to decide if we want to have a longer table.
@nkuehn what do you think?

File: packages/ui-kit/src/components/rss-feeds.js
Comment (TimonRey) [reply packages/ui-kit/src/components/rss-feeds.js:47]: That was always the case so I assumed that's already sorted 🤔

File: packages/ui-kit/src/components/rss-feeds.js
Comment (TimonRey) [reply packages/ui-kit/src/components/rss-feeds.js:27]: 👍

File: packages/ui-kit/src/components/rss-feeds.js
Comment (TimonRey) [reply packages/ui-kit/src/components/rss-feeds.js]: 👍

File: packages/ui-kit/src/components/rss-feeds.js
Comment (TimonRey) [reply packages/ui-kit/src/components/rss-feeds.js]: I moved it now into a new function outside of the component 👍

File: packages/ui-kit/src/components/rss-feeds.js
Comment (TimonRey) [reply packages/ui-kit/src/components/rss-feeds.js]: I reduced it into two now...
a8ed1dc

File: packages/ui-kit/src/components/rss-feeds.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.js:21]: This function has some "complex" logic. It's better to provide some unit tests here, to make sure the logic is solid.

**Comment by nkuehn** [comment]
> So just I can be sure, is this the UI Adina/Sven expected?

the heading above it would have to be removed for sure on the home page but sure, we need feedback here.

File: packages/ui-kit/src/components/rss-feed-table.js
Comment (nkuehn) [inline packages/ui-kit/src/components/rss-feed-table.js:68]: Please do not use fixed pixel widths but rather let the browser optimize and prevent word wrapping the first two columns by giving them  `white-space: nowrap;`.  That's generally better layout and also more robust in small screen widths.  

The preview layout uses too much white space in the first two columns while still breaking the third into multiple lines.

**Review by nkuehn — CHANGES_REQUESTED** [review]

**Comment by nkuehn** [comment]
@TimonRey  please show @Shecki  the result before merging.  

On that occasion: I have received feedback from Kelly that it would be good if there were a way to navigate to all release notes from the release note feed on the home page (bit of a UX issue now that we're merging them).

File: packages/ui-kit/src/components/rss-feed-table.js
Comment (TimonRey) [reply packages/ui-kit/src/components/rss-feed-table.js:68]: 👍

**Comment by TimonRey** [comment]
@nkuehn I discussed the design with @Shecki agreed to these changes. I also implemented links to navigate to the release notes of each "area"
c48b243

File: packages/ui-kit/src/components/rss-feed-table.js
Comment (TimonRey) [inline packages/ui-kit/src/components/rss-feed-table.js:65]: FYI: I wasn't sure about the naming here...I asked the docs team and they were for "Area". But it's still in discussion. Let me know what you think about it

File: packages/ui-kit/src/components/rss-feeds.js
Comment (TimonRey) [reply packages/ui-kit/src/components/rss-feeds.js]: 👍 
4c633a2

File: packages/ui-kit/src/components/rss-feeds.js
Comment (TimonRey) [reply packages/ui-kit/src/components/rss-feeds.js]: Hmm let me know if you would add some other unit tests as well...
4c633a2

**Comment by TimonRey** [comment]
I removed now the grey head column
1fc8635

File: packages/gatsby-theme-docs/gatsby-config.js
Comment (nkuehn) [reply packages/gatsby-theme-docs/gatsby-config.js:265]: Yes, this is correct. The number must match the maximum number we want to show in the feed because worst case all 10 last releases to be shown in the aggregated table are from one specific feed so all feeds must provide the maximum number of entries that the aggregated list would show.

**Review by nkuehn — CHANGES_REQUESTED** [review]
Hi, the UX is fine now.  But there are some visual glitches remaining. 

1. Please refer to the API Types for the exact styling of the outer borders (try to re-use the design tokens please  (e.g.   https://commercetools-docs-kit.commercetools.vercel.app/api-docs-smoke-test/types/examples.  The current implementation is lighter and nearly invisible at the top.  
2. The border styling flickers to a different style when hovering over the links (e.g. the dates). That's not intended, there should be no visual changes to the table when hovering content. 
3.

File: packages/ui-kit/src/components/rss-feeds.spec.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.spec.js:3]:
```suggestion
describe('transformData', () => {
```

File: packages/ui-kit/src/components/rss-feeds.spec.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.spec.js:7]:
```suggestion
  it('should limit the items with the newest of the oldest feed dates', () => {
```

File: packages/ui-kit/src/components/rss-feeds.spec.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.spec.js:12]:
```suggestion
  it('should show only items with a newer date sorted', () => {
```

File: packages/ui-kit/src/index.js
Comment (emmenko) [inline packages/ui-kit/src/index.js:8]: Why do we need to export this?

**Review by emmenko — COMMENTED** [review]
Thanks for adding a test case.

File: packages/ui-kit/src/index.js
Comment (TimonRey) [reply packages/ui-kit/src/index.js]: Sry...that is outdated 👍

File: packages/ui-kit/src/components/rss-feeds.spec.js
Comment (TimonRey) [reply packages/ui-kit/src/components/rss-feeds.spec.js]: Right, thx 👍

**Comment by emmenko** [comment]
Can you fix the build before I do a final review?

File: websites/docs-smoke-test/src/content/components/rss-feeds.mdx
Comment (emmenko) [inline websites/docs-smoke-test/src/content/components/rss-feeds.mdx:28]: We should pass a list, not a string.

Also, I think that the title can be extracted from the RSS feed, so we also don't need to explicitly pass it.

```suggestion
<RssFeeds dataSources={[
    "https://docs.commercetools.com/api/releases/feed.xml",
    "https://docs.commercetools.com/merchant-center/releases/feed.xml",
    "https://docs.commercetools.com/custom-applications/releases/feed.xml",
    "https://docs.commercetools.com/import-api/releases/feed.xml"
  ]}
/>
```

**Review by emmenko — CHANGES_REQUESTED** [review]

File: websites/docs-smoke-test/src/content/components/rss-feeds.mdx
Comment (TimonRey) [reply websites/docs-smoke-test/src/content/components/rss-feeds.mdx]: The `title` prop from the RSS feed is a whole sentence...but I was able to extract the feed name from it
d0e13bb

Comment (emmenko) [reply websites/docs-smoke-test/src/content/components/rss-feeds.mdx]: The `title` is defined by us here https://github.com/commercetools/commercetools-docs-kit/blob/2678b753d4cc9cbccd344edb6377672c7f51cef9/packages/gatsby-theme-docs/gatsby-config.js#L245

So you should be able to "reverse" that and extract the correct value.

Comment (TimonRey) [reply websites/docs-smoke-test/src/content/components/rss-feeds.mdx]: You mean by accessing the `siteMetadata`?

File: packages/ui-kit/src/components/rss-feeds.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.js:13]:
```suggestion
    const feedName = feedData.title.replace(/^commercetools (.*) Release Notes$/, "$1");
```

image

File: websites/docs-smoke-test/src/content/components/rss-feeds.mdx
Comment (emmenko) [reply websites/docs-smoke-test/src/content/components/rss-feeds.mdx]: Like this: https://github.com/commercetools/commercetools-docs-kit/pull/743#discussion_r522085954

File: packages/ui-kit/src/components/rss-feeds.js
Comment (TimonRey) [reply packages/ui-kit/src/components/rss-feeds.js]: 👍

File: jest.text.config.js
Comment (emmenko) [inline jest.text.config.js:9]: 💯

File: packages/ui-kit/src/components/rss-feeds.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.js:18]:
```suggestion
      const releaseNoteUrl = url.replace(/\/feed.xml$/, '');
```

File: packages/ui-kit/src/components/rss-feeds.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.js:85]:
```suggestion
        hasMultipleSources={props.dataSources.length > 1}
```

File: packages/ui-kit/src/components/rss-feeds.js
Comment (emmenko) [inline packages/ui-kit/src/components/rss-feeds.js:94]:
```suggestion
  dataSources: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
```

**Review by emmenko — COMMENTED** [review]
Looking good now! 🥳

File: jest.text.config.js
Comment (TimonRey) [reply jest.text.config.js:9]: Finally 😅

**Review by emmenko — APPROVED** [review]
Thanks.

I'll let @nkuehn do a final review and approve the Percy snapshots (they look good to me).

File: .changeset/calm-bulldogs-repair.md
Comment (nkuehn) [inline .changeset/calm-bulldogs-repair.md:2]: nitpick: this is not a patch but a minor release (a new feature).   

If I'm overzealous it's even a major because the API of the component changed.

**Review by nkuehn — APPROVED** [review]
Thanks, this looks very good now. 

Ideally, we would add the component to the theme's README, it's a sort of hidden feature at the moment.

File: .changeset/calm-bulldogs-repair.md
Comment (TimonRey) [reply .changeset/calm-bulldogs-repair.md]: I changed it to a minor now 👍

**Pull request merged**
